### PR TITLE
IDGEN-62 Set up module for REST webservices calls

### PR DESCRIPTION
**Pull request by adamsdenniskariuki**
# JIRA TICKET NAME:

[IDGEN-62 Set up module for REST web services calls](https://issues.openmrs.org/browse/IDGEN-62)

# SUMMARY:

Make the module ready for REST calls.

### CODE REVIEW

**Comment by cnagadya** [comment]
Looks good but if I am to nitpick a little, the first two files are missing the ending new line.
Also curious about the naming of **IdgenUIController.java** file, is it just a controller for the UI or for the entire OWA?

File: omod/src/main/resources/config.xml [hunk 1]
```diff
@@ -13,6 +13,7 @@
 	<activator>@MODULE_PACKAGE@.IdgenModuleActivator</activator>
 	<require_modules>
 		<require_module>org.openmrs.module.webservices.rest</require_module>
+		<require_module version="2.0.1">org.openmrs.module.idgen</require_module>
```
Comment (dkayiwa) [inline omod/src/main/resources/config.xml]: The idgen module cannot require itself. You can read more about that from: https://wiki.openmrs.org/display/docs/Module+Config+File

**Comment by dkayiwa** [comment]
@cnagadya that is a very good point 👍 
I would name it something like IdgenRestController

File: omod/src/main/java/org/openmrs/module/idgen/web/controller/IdgenUIController.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.stereotype.Controller;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceController;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+@RequestMapping("/rest/" + RestConstants.VERSION_1 + IdgenUIController.OWA_NAMESPACE)
+public class IdgenUIController extends MainResourceController {
+
+ public static final String OWA_NAMESPACE = "/owa";
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/idgen/web/controller/IdgenUIController.java]: The namespace would be idgen

Comment (adamsdenniskariuki) [reply omod/src/main/java/org/openmrs/module/idgen/web/controller/IdgenUIController.java]: Fixed.

File: omod/src/main/resources/config.xml (hunk 1 above)
Comment (adamsdenniskariuki) [reply omod/src/main/resources/config.xml]: Fixed. The <require_module> tag is for external modules needed in the current module.

**Comment by dkayiwa** [comment]
You are still missing the require_module element in your config.xml

**Comment by dkayiwa** [comment]
The link to the ticket has wrong text.

**Pull request merged**